internal/models/user.go

```diff
@@ -0,0 +1,32 @@
+package models
+
+import (
+	"errors"
+	"github.com/jackc/pgx/v5/pgconn"
+	"gorm.io/gorm"
+	"time"
+)
+
+// User is a GORM user model.
+type User struct {
+	ID        uint      `gorm:"primaryKey" json:"id"`
+	Email     string    `gorm:"unique" json:"email"`
+	CreatedAt time.Time `json:"created_at"`
+}
+
+// Create creates a new User record.
+func (u *User) Create(email string) (*User, error) {
```

Comment:
i ll better have one file for model and methods move to another file like user_repository